docs/UserGuide.md

```diff
@@ -126,25 +134,43 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
 Examples:
 * `find John` returns `john` and `John Doe`
 * `find alex david` returns `Alex Yeoh`, `David Li`<br>
-  ![result for 'find alex david'](images/findAlexDavidResult.png)
+  <!--- ![result for 'find alex david'](images/findAlexDavidResult.png) --->
+
+### Filtering residents by field : `filter`
+
+Shows a list of residents whose fields match the input keywords.
+
+Format: `filter [/specifier] KEY:VALUE [ADDITIONAL_KEYS:ADDITIONAL VALUES]`
+
+* The default specifier, if not specified by the user is `/all`.
+* Currently, only two specifiers are supported:
+  * `/all` returns a resident if it fulfills **all** of the specified keywords (Logical AND operation).
+  * `/any` returns a resident if it fulfills **any** of the specified keywords (Logical OR operation).
+* Repeated keys are not permitted for `/all` specifier, e.g. `filter /all house:Draco house:Aquila` will not work.
+* Valid keys are those included when [adding](#adding-a-resident--add) or [editing](#editing-an-existing-resident--edit) a resident e.g. `NAME, PHONE_NUMBER, EMAIL, FLOOR, UNIT, and TAG`
+
+Examples:
+* `filter /all house:Draco gender:Male` returns residents who are in Draco house, **and** are Male.
+* `filter /any house:Draco house:Aquila` returns residents belonging to either `Draco` **or** `Aquila` house.
+* `filter major:ComputerScience` returns residents who are in ComputerScience
```

Comment:
Let's change to short hand `CS` instead. We will include the different short hands in the **Format for resident fields** section.

Reply:
Let's remove this line, since we are no longer implementing major as a required field.